### PR TITLE
Show small header on ingress panels when the sidebar is hidden

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Show small header with menu button on mobile or when the sidebar is hidden.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by noxhirsch** [comment]
Is it maybe possible to always show the header only on mobile? On desktop it's just wasted space and annoying (on mobile even more, but I get the point).

**Comment by balloob** [comment]
Maybe we can make it depend on if the sidebar is visible ? And would it be possible to make it thinner, maybe like how it's on the iPhone when you open another app?
AD9D5456-5F42-46DA-99FB-45F6728D5354 2

**Comment by bramkragten** [comment]
We can make it depend on if the sidebar is visible like we do on the config panel. Not sure about the smaller toolbar, it looks a bit weird and not sure if the touch target still works well on mobile...

image

File: hassio/src/hassio-main.ts [hunk 1]
```diff
@@ -143,7 +124,7 @@ class HassioMain extends ProvideHassLitMixin(HassRouterPage) {
       });
     });
 
-    makeDialogManager(this, document.body);
+    makeDialogManager(this, this);
```
Comment (bramkragten) [inline hassio/src/hassio-main.ts:127]: This should be done differently as it now interferes with the logic of `HassRouterPage` to add elements, we should probably add an element in between, like `<hassio>`: handles dialogs -> `<hassio-main>`: router

File: hassio/src/hassio-router.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    super.firstUpdated(changedProps);
+    this.addEventListener("hass-api-called", (ev) => this._apiCalled(ev));
+  }
+
+  protected updatePageEl(el) {
+    // the tabs page does its own routing so needs full route.
+    const route = el.nodeName === "HASSIO-PANEL" ? this.route : this.routeTail;
+
+    if ("setProperties" in el) {
+      // As long as we have Polymer pages
```
Comment (balloob) [inline hassio/src/hassio-router.ts:90]: We don't have these anymore?

File: src/layouts/hass-subpage.ts [hunk 3]
```diff
@@ -12,29 +12,44 @@ import { classMap } from "lit-html/directives/class-map";
...
 
 @customElement("hass-subpage")
 class HassSubpage extends LitElement {
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
   @property()
   public header?: string;
 
+  @property({ type: Boolean })
+  public rootnav = false;
```
Comment (balloob) [inline src/layouts/hass-subpage.ts:25]: How can a subpage be a rootnav?

File: hassio/src/ingress-view/hassio-ingress-view.ts [hunk 4]
```diff
@@ -26,20 +29,40 @@ class HassioIngressView extends LitElement {
...
   @internalProperty() private _addon?: HassioAddonDetails;
 
+  @property({ type: Boolean })
+  public narrow = false;
+
   protected render(): TemplateResult {
     if (!this._addon) {
       return html` <hass-loading-screen></hass-loading-screen> `;
     }
+    const inConfigPanel = location.pathname.startsWith("/hassio/ingress");
```
Comment (balloob) [inline hassio/src/ingress-view/hassio-ingress-view.ts:39]: Let's add a property to indicate this. This component should not be aware of what URLs it is mounted.

File: hassio/src/ingress-view/hassio-ingress-view.ts [hunk 5]
```diff
@@ -87,6 +135,12 @@ class HassioIngressView extends LitElement {
         height: 100%;
         border: 0;
       }
+
+      hass-subpage.config {
+        --app-header-background-color: var(--sidebar-background-color);
```
Comment (balloob) [inline hassio/src/ingress-view/hassio-ingress-view.ts:140]: Seems weird that we use the sidebar background for this? I think that it is better suited to stay primary color, as that makes it blend into the phone header, which is also styled like the theme.

File: hassio/src/ingress-view/hassio-ingress-view.ts [hunk 6]
```diff
@@ -56,27 +79,52 @@ class HassioIngressView extends LitElement {
...
+      return;
     }
+
+    if (!(await createSessionPromise)) {
+      await showAlertDialog(this, {
+        text: "Unable to create an Ingress session",
+        title: addon.name,
+      });
+      // history.back();
+      // return;
```
Comment (balloob) [inline hassio/src/ingress-view/hassio-ingress-view.ts:124]: Stale comment.

File: hassio/src/ingress-view/hassio-ingress-view.ts [hunk 7]
```diff
@@ -26,20 +29,40 @@ class HassioIngressView extends LitElement {
...
+      !this.narrow &&
+      this.hass.dockedSidebar !== "always_hidden"
+    ) {
+      return html` <iframe src=${this._addon.ingress_url!}></iframe> `;
+    }
 
     return html`
-      <hass-subpage .header=${this._addon.name} hassio>
-        <iframe src=${this._addon.ingress_url}></iframe>
+      <hass-subpage
```
Comment (balloob) [inline hassio/src/ingress-view/hassio-ingress-view.ts:50]: I think that we shouldn't use a hass-subpage. Just a div should be enough. 

I still think that we should be able to use the look of iOS. Maybe we can make it just 40px high, so it is still valid touch target, but it doesn't take up too much height. And just make the iFrame height(100% - 40px).

File: hassio/src/ingress-view/hassio-ingress-view.ts (hunk 5 above)
Comment (bramkragten) [reply hassio/src/ingress-view/hassio-ingress-view.ts:140]: This is just for when it is in the config panel, so it matches the color of the config header

**Review by balloob — APPROVED** [review]

**Review by ludeeus — APPROVED** [review]

**Pull request merged**